pkg/webservice/webservice.go

```diff
@@ -43,7 +46,7 @@ type WebService struct {
 func newRouter() *httprouter.Router {
 	router := httprouter.New()
 	for _, webRoute := range webRoutes {
-		handler := loggingHandler(webRoute.HandlerFunc, webRoute.Name)
+		handler := gzipHandler(loggingHandler(webRoute.HandlerFunc, webRoute.Name))
```

Comment:
I don't think we should unconditionally GZIP every response, only those that might be large. I'd suggest adding a "Compress" boolean to the route object in routes.go and use that to determine whether to add the gzip handler.

Reply:
For example, none of the /debug routes should have it. It should probably not be on anything that typically results in < 10 KB of data.